src/PhpSpreadsheet/Reader/Csv.php

```diff
@@ -531,4 +532,63 @@ public function canRead($pFilename)
 
         return in_array($type, $supportedTypes, true);
     }
+
+    private static function guessEncodingTestNoBom(string &$encoding, string &$contents, string $compare, string $setEncoding): void
+    {
+        if ($encoding === '') {
+            $pos = strpos($contents, $compare);
+            if ($pos !== false && $pos % strlen($compare) === 0) {
+                $encoding = $setEncoding;
+            }
+        }
+    }
+
+    private static function guessEncodingNoBom(string $filename): string
+    {
+        $encoding = '';
+        $contents = file_get_contents($filename);
+        self::guessEncodingTestNoBom($encoding, $contents, "\x00\x00\x00\x0a", 'UTF-32BE');
+        self::guessEncodingTestNoBom($encoding, $contents, "\x0a\x00\x00\x00", 'UTF-32LE');
+        self::guessEncodingTestNoBom($encoding, $contents, "\x00\x0a", 'UTF-16BE');
+        self::guessEncodingTestNoBom($encoding, $contents, "\x0a\x00", 'UTF-16LE');
+        if ($encoding === '' && 1 == preg_match('//u', $contents)) {
+            $encoding = 'UTF-8';
+        }
+
+        return $encoding;
+    }
+
+    private static function guessEncodingTestBom(string &$encoding, string $first4, string $compare, string $setEncoding): void
+    {
+        if ($encoding === '') {
+            if ($compare === substr($first4, 0, strlen($compare))) {
+                $encoding = $setEncoding;
+            }
+        }
+    }
+
+    private static function guessEncodingBom(string $filename): string
+    {
+        $encoding = '';
+        $first4 = file_get_contents($filename, false, null, 0, 4);
+        if ($first4 !== false) {
+            self::guessEncodingTestBom($encoding, $first4, self::UTF8_BOM, 'UTF-8');
```

Comment:
Yes, I omitted doing that because each of the other constants would only be used once, but I agree that that's kind of a lazy approach. I have now added them and pushed the resulting change.